flowy/src/lib.rs

```diff
@@ -67,35 +67,40 @@ pub fn get_dir(path: &Path, solar_filter: &str) -> Result<Vec<String>, Box<dyn E
     files.sort();
     Ok(files)
 }
+
 /// Does esentially the same thing as generate_config
 /// Only runs when sunrise and sunset times
 /// need to be accounted for
 /// Takes lat and long of a location along with the wallpaper path
 pub fn generate_config_solar(path: &Path, lat: f64, long: f64) -> Result<(), Box<dyn Error>> {
     let mut day_walls = get_dir(path, "DAY")?;
     let night_walls = get_dir(path, "NIGHT")?;
-    let unixtime = DateTime::timestamp(&chrono::offset::Utc::now()) as f64;
+    let unixtime = DateTime::timestamp(&Utc::now()) as f64;
 
     let tt = solar::Timetable::new(unixtime, lat, long);
-    let (sunrise, sunset) = tt.get_sunrise_sunset();
-    println!("Sunrise {} Sunset {}", sunrise, sunset);
-    let mut sunrise = solar::time_to_mins(sunrise);
-    let mut sunset = solar::time_to_mins(sunset);
-
-    let day_len = ((sunset as i32 - sunrise as i32) % 1440) as u32;
-    let night_len = 1440 - day_len;
-    let day_div = day_len / day_walls.len() as u32;
-    let night_div = night_len / night_walls.len() as u32;
+    let sunrise = tt.get(&solar::SolarTime::Sunrise).unwrap().round() as i64;
+    let sunset = tt.get(&solar::SolarTime::Sunset).unwrap().round() as i64;
+
+    // Day length in seconds
+    let day_len = sunset - sunrise;
+    // Night length in seconds
+    let night_len = 86400 - day_len;
```

Comment:
Add a ``` % 86400``` at the end of the paranthesized statements so that we do not have to worry about the values being more that the number of seconds in a day